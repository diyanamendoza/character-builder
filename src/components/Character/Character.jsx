import React from 'react';

export default function Character({ head, torso, feet }) {
  return (
    <div
      className="character"
      style={{ display: 'flex', flexDirection: 'column', gap: 10 }}
    >
      <div
        className="head"
        style={{
          backgroundImage: `url(./${head}-head.png)`,
          backgroundSize: 'cover',
          height: 180,
          width: 320,
          backgroundRepeat: 'no-repeat',
        }}
      ></div>
      <div
        className="torso"
        style={{
          backgroundImage: `url(./${torso}-torso.png)`,
          backgroundSize: 'cover',
          height: 280,
          width: 320,
          backgroundRepeat: 'no-repeat',
        }}
      ></div>
      <div
        className="feet"
        style={{
          backgroundImage: `url(./${feet}-feet.png)`,
          backgroundSize: 'cover',
          height: 240,
          width: 320,
          backgroundRepeat: 'no-repeat',
        }}
      ></div>
    </div>
  );
}
