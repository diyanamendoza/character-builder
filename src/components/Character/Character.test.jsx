import { render } from '@testing-library/react';
import Character from './Character.jsx';

it('should render character component', () => {
  const { container } = render(
    <Character head="deadpool" torso="deadpool" feet="deadpool" />
  );
  expect(container).toMatchSnapshot();
});
