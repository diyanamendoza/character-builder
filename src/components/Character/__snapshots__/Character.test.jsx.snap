// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render character component 1`] = `
<div>
  <div
    class="character"
    style="display: flex; flex-direction: column; gap: 10px;"
  >
    <div
      class="head"
      style="background-image: url(./deadpool-head.png); background-size: cover; height: 180px; width: 320px; background-repeat: no-repeat;"
    />
    <div
      class="torso"
      style="background-image: url(./deadpool-torso.png); background-size: cover; height: 280px; width: 320px; background-repeat: no-repeat;"
    />
    <div
      class="feet"
      style="background-image: url(./deadpool-feet.png); background-size: cover; height: 240px; width: 320px; background-repeat: no-repeat;"
    />
  </div>
</div>
`;
